internal/controllers/machineset/machineset_controller.go

```diff
@@ -678,6 +681,28 @@ func (r *Reconciler) updateExternalObject(ctx context.Context, obj client.Object
 	return nil
 }
 
+func (r *Reconciler) getOwnerMachineDeployment(ctx context.Context, machineSet *clusterv1.MachineSet) (*clusterv1.MachineDeployment, error) {
+	mdName := machineSet.Labels[clusterv1.MachineDeploymentNameLabel]
```

Comment:
Are there really cases where this label is not set? I would suggest to just return an error if it's not set.

(we always set it in computeMachineSet in the MD controller and that should be enough to enforce it is always there)

Reply:
True, but why not fallback to looking to the owners?

Reply:
We can, but seems like unnecessary code. If we would check in all places that we use the label as a fallback the ownerRef as well this would be a lot of noise.

If I see correctly we use this label in a lot of places and this would be the first or one of the very first places where we fallback to ownerRefs.

If we think it's necessary we should probably write a util and use it everywhere

Reply:
(Just seems strange to rely on it in some places and not in others)

Reply:
reworked this logic